File: code/2153. Filtering Operators, Part 1 2-002-filtering-operators-part-1/2-002-filtering-operators-part-1/pre-recording/end/src/main/kotlin/SupportCode.kt
fun exampleOf(description: String, action: () -> Unit) {
  println("\n--- Example of: $description ---")
  action()
}

const val landOfDroids = "Land of Droids"
const val wookieWorld = "Wookie World"
const val detours = "Detours"

const val mayTheOdds = "And may the odds be ever in your favor"
const val liveLongAndProsper = "Live long and prosper"
const val mayTheForce = "May the Force be with you"

data class Movie(val title: String, val rating: Int)

val episodeI = Movie("The Phantom Menace", 55)
val episodeII = Movie("Attack of the Clones", 66)
val episodeIII = Movie("Revenge of the Sith", 79)
val rogueOne = Movie("Rogue One", 85)
val episodeIV = Movie("A New Hope", 93)
val episodeV = Movie("The Empire Strikes Back", 94)
val episodeVI = Movie("Return Of The Jedi", 80)
val episodeVII = Movie("The Force Awakens", 93)
val episodeVIII = Movie("The Last Jedi", 91)
val tomatometerRatings = listOf(
    episodeI, episodeII, episodeIII, rogueOne, episodeIV, episodeV, episodeVI, episodeVII, episodeVIII)

enum class Droid {
  C3PO, R2D2
}

File: code/1721. Filtering Operators RxSwift-2-02/RxSwift-2-02/12-filtering-operators-starter/RxSwiftPlayground/RxSwiftPlayground.playground/Sources/SupportCode.swift
import Foundation

public func example(of description: String, action: () -> Void) {
  print("\n--- Example of:", description, "---")
  action()
}

public let landOfDroids = "Land of Droids"
public let wookieWorld = "Wookie World"
public let detours = "Detours"

public let mayTheOdds = "And may the odds be ever in your favor"
public let liveLongAndProsper = "Live long and prosper"
public let mayTheForce = "May the Force be with you"

public let episodeI = (title: "The Phantom Menace", rating: 55)
public let episodeII = (title: "Attack of the Clones", rating: 66)
public let episodeIII = (title: "Revenge of the Sith", rating: 79)
public let rogueOne = (title: "Rogue One", rating: 85)
public let episodeIV = (title: "A New Hope", rating: 93)
public let episodeV = (title: "The Empire Strikes Back", rating: 94)
public let episodeVI = (title: "Return Of The Jedi", rating: 80)
public let episodeVII = (title: "The Force Awakens", rating: 93)
public let episodeVIII = (title: "The Last Jedi", rating: 91)
public let tomatometerRatings = [episodeI, episodeII, episodeIII, rogueOne, episodeIV, episodeV, episodeVI, episodeVII, episodeVIII]

public enum Droid {
  case C3PO, R2D2
}
